Userland/Libraries/LibCompress/Zlib.cpp

```diff
@@ -122,42 +131,48 @@ size_t ZlibCompressor::write(ReadonlyBytes bytes)
     return n_written;
 }
 
-bool ZlibCompressor::write_or_error(ReadonlyBytes bytes)
+bool ZlibCompressor::is_eof() const
 {
-    if (write(bytes) < bytes.size()) {
-        set_fatal_error();
-        return false;
-    }
+    return false;
+}
 
-    return true;
+bool ZlibCompressor::is_open() const
+{
+    return m_output_stream->is_open();
 }
 
-void ZlibCompressor::finish()
+void ZlibCompressor::close()
+{
+}
+
+ErrorOr<void> ZlibCompressor::finish()
 {
     VERIFY(!m_finished);
 
     if (is<DeflateCompressor>(m_compressor.ptr()))
         static_cast<DeflateCompressor*>(m_compressor.ptr())->final_flush();
 
     NetworkOrdered<u32> adler_sum = m_adler32_checksum.digest();
-    m_output_stream << adler_sum;
+    TRY(m_output_stream->write(adler_sum.bytes()));
 
     m_finished = true;
+
+    return {};
 }
 
 ErrorOr<ByteBuffer> ZlibCompressor::compress_all(ReadonlyBytes bytes, ZlibCompressionLevel compression_level)
 {
-    DuplexMemoryStream output_stream;
-    auto zlib_stream = TRY(ZlibCompressor::construct(output_stream, compression_level));
+    auto output_stream = make<Core::Stream::AllocatingMemoryStream>();
```

Comment:
This also wants to call something fallible instead of `make<>()`.

Reply:
That one I can definitely fix though, especially since it stays for good. :^)